llvm/utils/repo/Makefile

```diff
@@ -65,24 +71,23 @@ libcompiler-rt.a:
 		-D CMAKE_CXX_COMPILER=$(LLVM_BUILD)/bin/clang++               \
 		-D CMAKE_C_COMPILER=$(LLVM_BUILD)/bin/clang                   \
 		-D utils_dir=$(LLVM_REPO)/llvm/utils/repo                     \
-		-D CMAKE_INSTALL_PREFIX=$(LLVM)                               \
+		-D CMAKE_INSTALL_PREFIX=$(CLANG_RT)                           \
 		-D musl_install=$(MUSL)                                       \
 		../compiler-rt &&                                             \
 	make -j 8 &&                                                          \
 	make install
 
-$(LLVM)/lib/linux/clang_rt.crtbegin-x86_64.o: libcompiler-rt.a
-$(LLVM)/lib/linux/clang_rt.crtend-x86_64.o: libcompiler-rt.a
-
-$(LLVM)/lib/linux/clang_rt.crtbegin-x86_64.o.elf: $(LLVM)/lib/linux/clang_rt.crtbegin-x86_64.o
-	repo2obj --repo=$(LLVM_REPO)/build_libcompiler-rt/libcompiler-rt.db $< -o $@
+$(CLANG_RT)/lib/linux/clang_rt.crtbegin-x86_64.o: libcompiler-rt.a
+	mv $@ $(CLANG_RT)/lib/linux/clang_rt.crtbegin-x86_64.t
+	repo2obj --repo=$(LLVM_REPO)/build_libcompiler-rt/libcompiler-rt.db \
+	$(CLANG_RT)/lib/linux/clang_rt.crtbegin-x86_64.t -o $@

```

Comment:
I still don’t understand why this .o file has a prerequisite of libcompiler-rt.a, especially since it doesn’t appear anywhere in the recipe. 

I think that this won’t work if libcompiler-rt.a exists but crtbegin does not. The mv will fail won’t it?

Reply:
Is there a reason you don’t do the .o → .t rename in the recipe for libcompiler-rt.a?

Reply:
Following the discussion over the phone, I have modified/updated the patch:
>
>Is there a reason you don’t do the .o → .t rename in the recipe for libcompiler-rt.a?
>
1. Modified in the commit https://github.com/SNSystems/llvm-project-prepo/commit/6fe161a249b9b9417a4b8be8eef2f92d4424de5a.

2. Use -rm instead of rm. (commit https://github.com/SNSystems/llvm-project-prepo/commit/910090e5976d3f50e3d6e946efb0a503cb7aa4c4).
3. Use the VPATH to search path for prerequisites.  (commit https://github.com/SNSystems/llvm-project-prepo/commit/8df89c703c7b27b997f98919fa1b7af0dd7957b0)
4. Using One Shell for all libraries' targets.  (commit https://github.com/SNSystems/llvm-project-prepo/commit/8a111ca96a51474143dac8e7dcfa568c0d7bca14).